### ISSUE TITLE
Querying using :recurse for _include and _revinclude params fails

### ISSUE THREAD

**Issue by yosbeleg89** [Aidbox/Issues#18]
When executing a query to Aidbox using the [:recurse](http://hl7.org/fhir/2018May/search.html#revinclude) modifier (FHIR version 3.3.0) the server is returning an error:

#### Request:
`GET /MedicationDispense?_include=MedicationDispense:authorizingPrescription&_include:recurse=MedicationRequest:prescriber`

#### Response:
`Status: 500`
```
resourceType: OperationOutcome
issue:
- {severity: fatal, diagnostics: "java.lang.Exception: {:status :error, :errors [{:type\
    \ :error, :message \"No search parameter for MedicationDispense.authorizingPrescription\"\
 ...
```
#### Request:
`GET /Patient?_id=pat1&_revinclude=Observation:subject&_include:recurse=performer`

#### Response:
`Status: 500`
```
resourceType: OperationOutcome
issue:
- {severity: fatal, diagnostics: "java.lang.Exception: {:status :error, :errors [{:type\
    \ :error, :message \"No search parameter for Patient.include\"}]}\n at
 ...
```

**Comment by abcdw** [comment]
Hi, @yosbeleg89 Aidbox doesn't support `:recurse` parameter and probably won't be.
One of the reasons: https://build.fhir.org/search.html doesn't contain `:recurse`.

Aidbox have some capabilities beyond FHIR spec, If you have some tricky use case, we can help you to figure out the solution.

**Comment by yosbeleg89** [comment]
Thanks @abcdw, I know that since version 3.5.0 it was changed to `:iterate` to achieve the same purpose, does Aidbox will support `:iterate`? We also think that the response should be a `400 Bad Request` instead of 500.

**Comment by abcdw** [comment]
Currently, we are working on the refactoring of search engine and documentation of implemented features. Not sure about the support of `:iterate`, but anyway we will write a documentation and how to achieve the same result (for example with custom search and `_query`).

You are right about response code, should be fixed during refactoring.

**Comment by abcdw** [comment]
I think that most of the search documentation will be written in 2-3 weeks and all available options will be clear from it.

**Comment by mlapshin** [comment]
`_include:iterate` is implemented in `0.3.3-140-g9b7aea5`, feel free to reopen this issue if there are still any problems with it

for now maximum inclusion (recursion) depth is 5

**Comment by yosbeleg89** [comment]
@mlapshin We are having problems with the `_include:iterate` param, it is not including the specified resource.

Query example:

```
/fhir/DiagnosticReport?_count=10&_totalMethod=count&subject=123&_include=DiagnosticReport:result&_include:iterate=Observation:has-member
```
In this case the Observations targeted by `has-member` search parameter are not included in the reponse.
